packages/nextjs/app/_components/Grants/LargeGrantItem.tsx

```diff
@@ -0,0 +1,102 @@
+import { useRef } from "react";
+import Link from "next/link";
+import { LargeGrantMilestonesModal } from "./LargeGrantMilestonesModal";
+import { formatUnits } from "viem";
+import { MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { Badge } from "~~/components/pg-ens/Badge";
+import { Button } from "~~/components/pg-ens/Button";
+import { LargeGrantProgressBar } from "~~/components/pg-ens/LargeGrantProgressBar";
+import { Address } from "~~/components/scaffold-eth";
+import { useAuthSession } from "~~/hooks/pg-ens/useAuthSession";
+import { PublicLargeGrant } from "~~/services/database/repositories/large-grants";
+import { LargeMilestone } from "~~/services/database/repositories/large-milestones";
+import { LargeStage } from "~~/services/database/repositories/large-stages";
+import { getFormattedDate } from "~~/utils/getFormattedDate";
+import { multilineStringToTsx } from "~~/utils/multiline-string-to-tsx";
+
+type GrantItemProps = {
+  grant: PublicLargeGrant;
+  latestsShownStatus: "all" | "approved";
+};
+
+export const LargeGrantItem = ({ grant, latestsShownStatus }: GrantItemProps) => {
+  const milestonesRef = useRef<HTMLDialogElement>(null);
+  const { isAdmin } = useAuthSession();
+
+  const latestStage =
+    latestsShownStatus === "approved"
+      ? (grant.stages.find(
+          stage => stage.status === "approved" || stage.status === "completed",
+        ) as unknown as LargeStage)
+      : grant.stages[0];
+
+  const showProposalTitle =
+    latestsShownStatus === "all" &&
+    latestStage.stageNumber === 1 &&
+    (latestStage.status === "proposed" || latestStage.status === "rejected");
+
+  const allMilestonesGrantAmount = grant.stages
+    .map(stage =>
```

Comment:
nit: I think it's possible with `flatMap` and one `reduce`

Reply:
Yeah! Changed! And changed the two below too.